### PR TITLE
binary-compatibility-validator 0.5.0 fork

### PR DESCRIPTION
**Pull request by CherryPerry**
Fork binary-compatibility-validator 0.5.0 and enable dump only for jvm target.

Most of the changes are just `git clone`. Did not want to fork it as a separate repo on GitHub.

### CODE REVIEW

File: tools/binary-compatibility-validator/src/main/kotlin/BinaryCompatibilityValidatorPlugin.kt [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+            }.all {
+                project.configureKotlinCompilation(it, extension, useOutput = true)
+            }
+        }
+
+        project.pluginManager.withPlugin("kotlin-multiplatform") {
+            if (project.name in extension.ignoredProjects) return@withPlugin
+            val kotlin = project.extensions.getByName("kotlin") as KotlinMultiplatformExtension
+            kotlin.targets.matching {
+                it.platformType == KotlinPlatformType.jvm
```
Comment (CherryPerry) [inline tools/binary-compatibility-validator/src/main/kotlin/BinaryCompatibilityValidatorPlugin.kt:62]: Removed `KotlinPlatformType.jvm` here.

**Comment by arkivanov** [comment]
Seems like we should comply with the Apache license's points `4.a` and `4.b`: https://github.com/Kotlin/binary-compatibility-validator/blob/master/LICENSE.TXT
Perhaps copy LICENSE.txt to the folded and also add some comments to the modified files.

**Review by arkivanov — CHANGES_REQUESTED** [review]
See the comment above

**Comment by CherryPerry** [comment]
Added notice e2d8047, LICENSE file is already provided.

**Review by arkivanov — APPROVED** [review]

**Pull request merged**